### DISCUSSION TITLE
Can I use html comments for jStachio delimiters in html templates?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [jstachio/jstachio#201] [General]
Is it possible to re-use HTML comments **()** for delimiters like show  here:
```
<!-- #names -->
<!-- . -->
<!-- /names-->

```
**This would make an HTML template look a lot cleaner and better when previewed in a browser!**

Reading the docs, it says the delimiters should be 2-char strings, unfortunately HTML comments are 3-char long.

I would even even use ```<!--|``` and ```|-->``` if possible., to better differentiate jStachio delimiters from HTML comments. 

Also, can any of the above to  be configured in **@JStacheConfig** rather than through the mechanism (=) descibed in the spec?

**Comment by agentgt** [comment]
Why did you close it?

Anyway it is valid that we should probably try to support three character delimiters.

**Comment by hrstoyanov** [comment]
I figured its is stupid, because consider this example using  as delimiters:
`<li>``</li>`
When you open the template in Chrome, it will not show you anything in the li (btw, you need to see the raw markdown, since Github also strips comments).


I think it is a valid request to be able to do >2 chars delimiters, either via JStacheConfig or the canonical way from the specs, so if you agree maybe we can added to the list of future enhancements?

**Comment by hrstoyanov** [comment]
... @agentgt Well, here is an idea that migth actually work - allow for 2 or more ways to specify delimiters, something like:

```java
@JStacheConfig(
     ...
     delimiters = {

           //note this pair is also HTML comment!
           @Delimiter( start="<!--|",   end="|-->"  ), 

          //default delimiters can still be used, not HTML comments, so it is visible in Chrome.
           @Delimiter( start="{{",   end="}}" ),

           //... and even more delimiter pairs can be defined, if needed.
          ...

   }
)
```


This way, you can be flexible with your html template, and it will render very cleanly, when previewed (unprocessed) in Chrome:

```html
<!-- This is an HTML comment, but it will not interfere with jStachio -->

<!--|#names|-->

<!--|#-first|--> 
<p>Names:</p>
<ul>
<!--|/-first|-->  

<li>{{! Example Name - shows in Chrome nicely, no effect on jStachio}} {{.}}</li>

<!--|#-last|-->  
</ul>
<!--|/-last|-->  

<!--| /names |-->
```

**Comment by agentgt** [comment]
The config part is fairly straightforward.

More than 2 chars is tricky and will take time.

**Comment by hrstoyanov** [comment]
Cool, thanks. I would say @JStacheConfig should be the only way to configure multiple  delimiters, don't bother with the Mustache way from inside the template.

**Reply by agentgt** [reply]
> don't bother with the Mustache way from inside the template.

I need to to support the spec but that is kind of orthogonal to the whole set delimiter external. Regardless of where delimiters are set it is a challenge to fix parsing for delimiters more than 2 characters long because of how the tokenizer works currently.
